### DISCUSSION TITLE
Accessing variables created under plugins to do data logging

### DISCUSSION THREAD

**Discussion by ZJLim** [TUDelft-CNS-ATM/bluesky#476] [General]
Hi, 

I cannot seem to access the new per-aircraft states that I have created in my plugins.
I want to log these new states while running the simulation.

When I use the stack command 'lsvar', I am able to see my plugin which is 'dist2fix'.
<img width="180" alt="image" src="https://github.com/TUDelft-CNS-ATM/bluesky/assets/57678432/95d36bbc-19cc-417c-b76f-2505ce8e7da8">

Next, when I use the stack command 'lsvar dist2fix', I managed to see the attributes that I have introduced which are 'dist2dest', 'dist2IAF' and 'STAR'.
<img width="266" alt="image" src="https://github.com/TUDelft-CNS-ATM/bluesky/assets/57678432/43836974-f937-4f56-bafd-ca3713e8b6dc">

However, when I use the stack command 'lsvar dist2fix.dist2dest', there is an error indicating that the variable dist2fix.dist2dest is not found.
<img width="366" alt="image" src="https://github.com/TUDelft-CNS-ATM/bluesky/assets/57678432/64e462c6-d82e-42f3-bf2c-cf5fe090b301">

Kindly advise on how I can access the new per-aircraft states that I have introduced in my plugins to do data logging. The snapshot below shows how I have initialised the class.

<img width="459" alt="image" src="https://github.com/TUDelft-CNS-ATM/bluesky/assets/57678432/def79a59-9b1d-4ab7-9607-b5a6b8604d37">

Update: My current workaround is I edited the traffic.py file directly by adding the attributes in it. But I hope there exists another way in which I can do everything within the plugin file instead of editing the source code.

Thank you.
Zhi Jun

**Comment by jooste** [comment]
Hi @ZJLim,

I think this should have worked, so it looks like you've found a bug! We'll pick this up after the summer holidays.

**Reply by ZJLim** [reply]
Hi @jooste,

Thank you for getting back to me so quickly. I look forward to the possible fix but in the meantime, I will add the new per-aircraft states in the traffic.py file to access them for data logging.
